src/main/java/org/jetbrains/bsp/bazel/server/bsp/BazelBspServerBuildManager.java

```diff
@@ -179,28 +186,95 @@ public BuildTarget getBuildTargetForRule(Build.Rule rule) {
               scalaVersion.substring(0, scalaVersion.lastIndexOf(".")),
               ScalaPlatform.JVM,
               classpath);
-      scalacClasspath.setJvmBuildTarget(getJVMBuildTarget());
+      scalacClasspath.setJvmBuildTarget(getJVMBuildTarget(rule));
     }
 
     return Optional.of(scalacClasspath);
   }
 
-  private JvmBuildTarget getJVMBuildTarget() {
-    // TODO(andrefmrocha): Properly determine jdk path
-    return new JvmBuildTarget(null, getJavaVersion());
+  private JvmBuildTarget getJVMBuildTarget(Build.Rule rule) {
+    Optional<String> javaHomePath = getJavaPath(rule);
+    return new JvmBuildTarget(javaHomePath.orElse(null), getJavaVersion().orElse(null));
   }
 
-  private String getJavaVersion() {
-    String version = System.getProperty("java.version");
-    if (version.startsWith("1.")) {
-      version = version.substring(0, 3);
-    } else {
-      int dot = version.indexOf(".");
-      if (dot != -1) {
-        version = version.substring(0, dot);
+  private Optional<String> getJavaPath(Build.Rule rule) {
+    List<String> traversingPath = Lists.newArrayList("$jvm", "$java_runtime", ":alias", "actual");
+    Optional<Build.Rule> jdkRule = traverseDependency(rule, traversingPath);
+
+    if (!jdkRule.isPresent()) {
+      return Optional.empty();
+    }
+
+    String jdkLocation = jdkRule.get().getLocation();
+    String localJdkPath = jdkLocation.substring(0, jdkLocation.indexOf("/BUILD"));
+    return Optional.of(localJdkPath);
+  }
```

Comment:
it can be done in the single call (something like that):

```suggestion
    List<String> traversingPath = Lists.newArrayList("$jvm", "$java_runtime", ":alias", "actual");
    return traverseDependency(rule, traversingPath)
       .map(Build.Rule::getLocation)
       .map(location -> location.substring(0, jdkLocation.indexOf("/BUILD"));
  }
```